pybop/costs/_likelihoods.py

```diff
@@ -352,3 +352,99 @@ def _evaluateS1(self, inputs: Inputs) -> tuple[float, np.ndarray]:
         total_gradient = dl + prior_gradient
 
         return posterior, total_gradient
+
+
+class LogPosterior(BaseCost):
```

Comment:
This class should merge with and replace `MAP`.

Reply:
Agreed, but let's do this in a new PR, as it's out of scope on this one. I'll open an issue.